servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -403,15 +439,16 @@ public Completable closeAsyncGracefully() {
 
         @SuppressWarnings("unchecked")
         private Completable doClose(final Function<? super C, Completable> closeFunction) {
-            return defer(() -> succeeded((List<C>) connectionsUpdater.getAndSet(this, CLOSED_LIST)))
-                    .flatMapCompletable(list -> mergeAllDelayError(list.stream().map(closeFunction)::iterator));
+            return defer(() -> succeeded(connectionsUpdater.getAndSet(this, CLOSED_ARRAY)))
+                    .flatMapCompletable(array -> mergeAllDelayError(
+                            Stream.of(array).map(conn -> closeFunction.apply((C) conn))::iterator));
         }
 
         @Override
         public String toString() {
             return "Host{" +
                     "address=" + address +
-                    ", removed=" + (connections == CLOSED_LIST) +
+                    ", removed=" + (connections == CLOSED_ARRAY) +
```

Comment:
While you are here, please add `serialVersionUID` for `StacklessNoAvailableHostException`

Reply:
done